server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -171,199 +162,111 @@ static void clearLoadedModules()
+                  extensionDir.getAbsolutePath()
+              )
+          );
         }
-
-        loader = new URLClassLoader(urls.toArray(new URL[urls.size()]), Initialization.class.getClassLoader());
-        loadersMap.put(coordinate, loader);
-      }
-      catch (Exception e) {
-        log.error(e, "Unable to resolve artifacts for [%s].", dependencyRequest);
-        throw Throwables.propagate(e);
+        extensionsToLoad[i++] = extensionDir;
       }
     }
-    return loader;
+    return extensionsToLoad == null ? new File[]{} : extensionsToLoad;
   }
 
-  public static DefaultTeslaAether getAetherClient(ExtensionsConfig config)
+  /**
+   * Find all the hadoop dependencies that should be loaded by druid
+   *
+   * @param hadoopDependencyCoordinates e.g.["org.apache.hadoop:hadoop-client:2.3.0"]
+   * @param extensionsConfig            ExtensionsConfig configured by druid.extensions.xxx
+   *
+   * @return an array of hadoop dependency files that will be loaded by druid process
+   */
+  public static File[] getHadoopDependencyFilesToLoad(
+      List<String> hadoopDependencyCoordinates,
+      ExtensionsConfig extensionsConfig
+  )
   {
-    /*
-    DefaultTeslaAether logs a bunch of stuff to System.out, which is annoying.  We choose to disable that
-    unless debug logging is turned on.  "Disabling" it, however, is kinda bass-ackwards.  We copy out a reference
-    to the current System.out, and set System.out to a noop output stream.  Then after DefaultTeslaAether has pulled
-    The reference we swap things back.
-
-    This has implications for other things that are running in parallel to this.  Namely, if anything else also grabs
-    a reference to System.out or tries to log to it while we have things adjusted like this, then they will also log
-    to nothingness.  Fortunately, the code that calls this is single-threaded and shouldn't hopefully be running
-    alongside anything else that's grabbing System.out.  But who knows.
-    */
-
-    List<String> remoteUriList = config.getRemoteRepositories();
-
-    List<Repository> remoteRepositories = Lists.newArrayList();
-    for (String uri : remoteUriList) {
-      try {
-        URI u = new URI(uri);
-        Repository r = new Repository(uri);
-
-        if (u.getUserInfo() != null) {
-          String[] auth = u.getUserInfo().split(":", 2);
-          if (auth.length == 2) {
-            r.setUsername(auth[0]);
-            r.setPassword(auth[1]);
-          } else {
-            log.warn(
-                "Invalid credentials in repository URI, expecting [<user>:<password>], got [%s] for [%s]",
-                u.getUserInfo(),
-                uri
-            );
-          }
-        }
-        remoteRepositories.add(r);
-      }
-      catch (URISyntaxException e) {
-        throw Throwables.propagate(e);
-      }
+    final File rootHadoopDependenciesDir = new File(extensionsConfig.getHadoopDependenciesDir());
```

Comment:
Similar here, it is not explicitly checked, but if ```rootHadoopDependenciesDir``` doesn't exist, and ```hadoopCoordinates``` or ```hadoopDependency``` is not empty, ISE is thrown.